arkouda/dataframe.py

```diff
@@ -56,37 +58,186 @@ def groupby_operators(cls):
 
 @groupby_operators
 class GroupBy:
-    """A DataFrame that has been grouped by a subset of columns"""
+    """
+    A DataFrame that has been grouped by a subset of columns
+
+    Parameters
+    ----------
+
+    gb_key_names : str or list(str), default=None
+        The column name(s) associated with the aggregated columns.
+
+    as_index : as_index, default=True
+        If True, interpret aggregated column as index
+        (only implemented for single dimensional aggregates).
+        Otherwise, treat aggregated column as a dataframe column.
+
+    Attributes
+    ----------
+    gb :    arkouda.groupbyclass.GroupBy
+        GroupBy object, where the aggregation keys are values of column(s) of a dataframe,
+        usually in preparation for aggregating with respect to the other columns.
+    df :    arkouda.dataframe.DataFrame
+        The dataframe containing the original data.
+    gb_key_names    :    str or list(str)
+        The column name(s) associated with the aggregated columns.
+    as_index : bool (default=True)
+        If True the grouped values of the aggregation keys will be treated as an index.
+
+    """
 
-    def __init__(self, gb, df):
+    def __init__(self, gb, df, gb_key_names=None, as_index=True):
         self.gb = gb
         self.df = df
-        for attr in ["nkeys", "size", "permutation", "unique_keys", "segments"]:
+        self.gb_key_names = gb_key_names
+        self.as_index = as_index
```

Comment:
we talked about adding `skipna` to the init. I think we might be able to just add it as a class attribute here and then pass it into `self.gb.aggregate` in `aggop`. I'm up for this to be in this PR if it's easy or looked into in a separate issue, whatever u prefer!